scipy/_lib/decorator.py

```diff
@@ -270,11 +270,7 @@ def decorator(caller, _func=None):
 
 # ####################### contextmanager ####################### #
 
-try:  # Python >= 3.2
-    from contextlib import _GeneratorContextManager
-except ImportError:  # Python >= 2.5
-    from contextlib import GeneratorContextManager as _GeneratorContextManager
-
+from contextlib import _GeneratorContextManager
```

Comment:
hmm, this seems like a recipe for sudden breakage - while we're touching it, any way we could avoid using the private attribute?

Reply:
Since decorator.py is vendored code, maybe best just update it to the latest version from upstream (which is 4.4.1 ATM, and ours is 4.0. ) instead

Reply:
I'm with @ev-br on this. Just try updating the vendored code to the latest release.

Reply:
I'm reverted this file.

Silly question, how do I know what code in numpy/scipy is vendor code?

Reply:
There's no specific list, only hints in various places. 

e.g. L31: `Decorator module, see https://pypi.python.org/pypi/decorator`
e.g. in the LICENSES_bundled.txt:

> Name: Decorator
Files: scipy/_lib/decorator.py
License: 2-clause BSD
  For details, see the header inside scipy/_lib/decorator.py

Reply:
Off the top of my head (not a definitive list): SuperLU is certainly vendored, qhull too. Pocketfft is active upstream. Not sure about shgo and trlib. Cephes we effectively forked, several fortran libs in interpolate, stats and special are not active upstream, so likely ok to patch if needed.